src/DocumentationExamples/DocumentationExamples.csproj

```diff
@@ -1,7 +1,8 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>net8.0</TargetFramework>
+    <TargetFramework>net9.0</TargetFramework>
+    <LangVersion>latest</LangVersion>
```

Comment:
I would prefer not to use a floating version since this means different things at different times on different machines. We could accidentally start using a new lang feature without explicitly intending to opt into depending on the latest tooling.

When using `preview` in the projects that need a specific language feature, there's no way around it, but that's a temporary state of things that we'll fix as soon as it's possible to replace `preview` with a concrete version and retain access to that feature. And important to me to add an XML comment above it explaining which language feature we're waiting on until we can move this to a concrete language version.

Also, I would prefer keeping all projects on the same langversion by only editing here, instead of adding LangVersion in any .csproj files:

https://github.com/shouldly/shouldly/blob/50e00451de5c3ecba99aac232db329a6104f41fc/src/Directory.Build.props#L4

Reply:
I did further testing and Preview 7 allows the use of langversion 13, so we don't need to set it to `preview` 🎉

Because of this, let's not touch langversion anywhere. The defaults should let us compile.

Reply:
AFAIK you'll only ever get a new C# version with a new major .NET version, so I don't think there's really much "accident" happening in that scenario. If you want repeatability global.json shouldn't roll-forward or allow previews so you always get the version that is in source control. The with Latest you don't have to remember to update it every year to get all the new shinies 😄

I didn't spot the D.B.props file as it wasn't in the root which is where I'm used to finding it, and there was a project-level setting so I just assumed that was the pattern here.

Reply:
I'm not quickly finding a link for it, but `latest` is discouraged by the C# language team. Rollforward of the SDK version makes the contribution experience so much smoother for both experienced folks and newer contributors. Since `latest` overrides the default language for the target framework when you have a later SDK, that's where the problem lies. Lastly, remembering to update langversion is usually not a hassle since the IDE prompts you to change it the moment you start to want it. And when no langversion is specified at all and the default is used based on TargetFramework, you don't have to update it at all unless you're after a preview feature.

I appreciate your thoughtfulness in approaching this!

Reply:
I did find the link. The advice is buried underneath the table in the <https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-options/language#langversion> section:

> [!IMPORTANT]
> The `latest` value is generally not recommended. With it, the compiler enables the latest features, even if those features depend on updates not included in the configured target framework. Without this setting, your project uses the compiler version recommended for your target framework. You can update the target framework to access newer language features.

This did kick off a conversation. Bill plans to make this recommendation more visible soon.